Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PolDiffILLReduction.py

```diff
@@ -0,0 +1,876 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+
+from mantid.api import FileProperty, MatrixWorkspaceProperty, MultipleFileProperty, \
+    NumericAxis, PropertyMode, Progress, PythonAlgorithm, WorkspaceGroup, WorkspaceGroupProperty, \
+    FileAction, AlgorithmFactory
+from mantid.kernel import Direction, EnabledWhenProperty, IntBoundedValidator, LogicOperator, \
+    PropertyCriterion, PropertyManagerProperty, StringListValidator
+
+from mantid.simpleapi import *
+
+from scipy.constants import physical_constants
+import numpy as np
+import math
+
+
+class PolDiffILLReduction(PythonAlgorithm):
+
+    _mode = 'Monochromatic'
+    _method_data_structure = None # measurement method determined from the data
+    _user_method = None
+    _instrument = None
+    _experimentProperties = None
+
+    _DEG_2_RAD =  np.pi / 180.0
+
+    @staticmethod
+    def _max_value_per_detector(ws):
+        max_values = np.zeros(mtd[ws].getItem(0).getNumberOfHistograms())
+        for entry in mtd[ws]:
+            for spectrum_no in range(entry.getNumberOfHistograms()):
```

Comment:
I think it would be faster here if you extract the data to numpy and use its functions to find the max instead of looping over the spectra.